styleguide-app/Examples/Button.elm

```diff
@@ -139,79 +147,73 @@ viewButtonExamples messages (State control) =
         { label = "Delete Something"
         , onClick = messages.showItWorked "delete"
         }
-    , Button.linkExternalWithTracking
-        (messages.showItWorked "linkExternalWithTracking clicked")
-        { size = Button.Medium
-        , style = Button.Secondary
-        , width = Button.WidthUnbounded
-        , label = "linkExternalWithTracking"
-        , icon = Nothing
-        , url = "#"
-        }
+    , Button.build
+        |> Button.withLabel "linkExternalWithTracking"
+        |> Button.unboundedWidth
+        |> Button.secondary
+        |> Button.onClick (messages.showItWorked "linkExternalWithTracking clicked")
+        |> Button.linkExternalWithTracking
+        |> Button.renderLink
```

Comment:
❓ I wonder what the tradeoffs are between a pipeline API like this and an "attributes" API like

```elm
Button.button
    [ Button.label "linkExternalWithTracking"
    , Button.unboundedWidth
    , Button.secondary
    , Button.onClick msg
    ]
```

I wonder if @stoeffel has thoughts about this?

Reply:
Ah, here's the thing from Stöffel that I was thinking of: https://github.com/NoRedInk/engblog/pull/68

Reply:
In general, I came to think that a function with many optional fields in its configuration is an indication that it might do too many things and splitting it up into more specific functions might be better.

In this case though, I think having an html-like api or pipeline api will be an improvement.
I wonder which api gives the better error messages. 
Two things I like about the html-like api:
* no need to render at the end of the pipeline
* it feels like html :-)

Reply:
I went with the builder-render pattern mainly because @BrianHicks has been enthusiastic about it recently. I definitely like stuff to feel like html tho!

Maybe I'll try out an alternate attributes approach and see what that feels like 🤔